pilot/pkg/config/file/store.go

```diff
@@ -400,15 +415,59 @@ func (s *KubeSource) parseChunk(r *collection.Schemas, name string, lineNum int,
 	pos := kube2.Position{Filename: name, Line: lineNum}
 	c, err := ToConfig(objMeta, schema, &pos, fieldMap)
 	if err != nil {
-		return kubeResource{}, err
+		return resources, err
 	}
-	return kubeResource{
-		schema: schema,
-		sha:    sha256.Sum256(yamlChunk),
-		config: c,
+	return []kubeResource{
+		{
+			schema: schema,
+			sha:    sha256.Sum256(yamlChunk),
+			config: c,
+		},
 	}, nil
 }
 
+type resourceYamlChunk struct {
+	lineNum   int
+	yamlChunk []byte
+}
+
+func extractResourceChunksFromListYamlChunk(chunk []byte) ([]resourceYamlChunk, error) {
+	chunks := make([]resourceYamlChunk, 0)
+	yamlChunkNode := yamlv3.Node{}
+	err := yamlv3.Unmarshal(chunk, &yamlChunkNode)
+	if err != nil {
+		return nil, fmt.Errorf("failed parsing yamlChunk: %v", err)
+	}
+	if len(yamlChunkNode.Content) == 0 {
+		return nil, fmt.Errorf("failed parsing yamlChunk: no content")
+	}
+	yamlNode := yamlChunkNode.Content[0]
+	var itemsInd int
+	for ; itemsInd < len(yamlNode.Content); itemsInd++ {
+		if yamlNode.Content[itemsInd].Kind == yamlv3.ScalarNode && yamlNode.Content[itemsInd].Value == "items" {
+			itemsInd++
+			break
+		}
+	}
+	if itemsInd >= len(yamlNode.Content) && yamlNode.Content[itemsInd].Kind != yamlv3.SequenceNode {
```

Comment:
if `itemsInd >= len` won't `yamlNode.Content[itemsInd]` be out of bounds?

@hanxiaop